src/main/scala/com/wavesplatform/state2/NgState.scala

```diff
@@ -1,38 +1,69 @@
 package com.wavesplatform.state2
 
+import java.util.concurrent.TimeUnit
+
+import cats.kernel.Monoid
+import com.google.common.cache.CacheBuilder
 import scorex.block.Block.BlockId
 import scorex.block.{Block, MicroBlock}
 import scorex.transaction.History.BlockMinerInfo
-import scorex.transaction.{AssetId, DiscardedMicroBlocks, Transaction}
+import scorex.transaction.{DiscardedMicroBlocks, Transaction}
+
+import scala.collection.mutable.{ListBuffer => MList, Map => MMap}
+
+class NgState(val base: Block, val baseBlockDiff: BlockDiff, val acceptedFeatures: Set[Short]) {
+
+  private val MaxTotalDiffs = 3
 
+  private val microDiffs: MMap[BlockId, (BlockDiff, Long)] = MMap.empty
+  private val micros: MList[MicroBlock] = MList.empty // fresh head
+  private val totalBlockDiffCache = CacheBuilder.newBuilder()
+    .maximumSize(MaxTotalDiffs)
+    .expireAfterWrite(10, TimeUnit.MINUTES)
```

Comment:
Just interesting, why 10 minutes? This period is much more than a distance between blocks.